device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java

```diff
@@ -132,17 +133,21 @@ public void open(Queue<DeviceClientConfig> deviceClientConfigs, ScheduledExecuto
                 }
                 else if (this.config.getAuthenticationType() == DeviceClientConfig.AuthType.X509_CERTIFICATE)
                 {
-                    if (this.config.isUseWebsocket())
-                    {
-                        //Codes_SRS_MQTTIOTHUBCONNECTION_34_027: [If this function is called while using websockets and x509 authentication, an UnsupportedOperationException shall be thrown.]
-                        throw new UnsupportedOperationException("X509 authentication is not supported over MQTT_WS");
-                    }
-
                     this.log.trace("MQTT connection will use X509 certificate based auth");
 
                     this.iotHubUserPassword = null;
                 }
 
+                if (this.config.getAuthenticationType() != DeviceClientConfig.AuthType.X509_CERTIFICATE)
+                {
+                    //X509 auth websocket should not include the iothub-no-client-cert flag
+                    this.webSocketQueryString = NO_CLIENT_CERT_QUERY_STRING;
```

Comment:
Do we need "this" any more? I know it is there in legacy code but this going forward if we don't need it why use?

Reply:
C# guidelines suggest avoiding "this." when possible, but Java guidelines do not

Reply:
Sorry about that, I added this comment thinking about C#